ci/cloudbuild/cloudbuild.yaml

```diff
@@ -0,0 +1,45 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+options:
+  machineType: 'N1_HIGHCPU_32'
+  diskSizeGb: '512'
+
+substitutions:
+  _DISTRO: "unknown"
+  _BUILD_NAME: "unknown"
+
+timeout: 3600s
+
+steps:
+- name: 'gcr.io/cloud-builders/docker'
+  entrypoint: 'bash'
+  args: [
+    '-c',
+    'docker pull us-central1-docker.pkg.dev/$PROJECT_ID/google-cloud-cpp-cloudbuild-docker/${_DISTRO}-image:tag1 || exit 0'
+  ]
+- name: 'gcr.io/cloud-builders/docker'
+  args: [
+    'build',
+    '-f', 'ci/cloudbuild/${_DISTRO}.Dockerfile',
+    '-t', 'us-central1-docker.pkg.dev/$PROJECT_ID/google-cloud-cpp-cloudbuild-docker/${_DISTRO}-image:tag1',
+    '--cache-from', 'us-central1-docker.pkg.dev/$PROJECT_ID/google-cloud-cpp-cloudbuild-docker/${_DISTRO}-image:tag1',
```

Comment:
ah... I see, you are caching from the previous push... neat.

Note how a separate PR that changes `fedora.Dockerfile` will dirty up your cache.  I think everything works, just maybe not as well as it could.

Reply:
Yes, I'm attempting to do some caching of the docker image. I'd like to switch to Kaniko, though I cannot figure out why it doesn't work. I filed https://github.com/GoogleContainerTools/kaniko/issues/1604 about that.

Maybe what we want is something like post-submit builds push built images. Pre-submit PR builds pull images and `--cache-from` them, but then do not push their images.

Reply:
Ack, it might be worth trying `:edge` for Kaniko instead of `:latest`.

Reply:
Huh. I'd tried that before and it didn't work. However, I tried again today with `:edge` AND I changed to `gcr.io` instead of using the artifact registry and it seems to work. So, cool!